bcipy/helpers/convert.py

```diff
@@ -0,0 +1,137 @@
+"""Functionality for converting the bcipy raw data output to other formats"""
+
+import logging
+import os
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import numpy as np
+from pyedflib import FILETYPE_EDFPLUS, EdfWriter
+
+from bcipy.helpers.load import load_json_parameters, read_data_csv
+from bcipy.helpers.triggers import read_triggers, trigger_durations
+
+
+def convert_to_edf(data_dir: str,
+                   edf_path: str = None,
+                   overwrite=False,
+                   use_event_durations=False) -> Path:
+    """ Converts BciPy raw_data to the EDF+ filetype using pyEDFlib.
+
+    Parameters
+    ----------
+    raw_data_dir - directory which contains the data to be converted. This
+        location must also contain a parameters.json configuration file.
+    output_path - optional path to write converted data; defaults to writing
+        a file named raw.edf in the data_dir.
+    overwrite - If True, the destination file (if it exists) will be overwritten.
+        If False (default), an error will be raised if the file exists.
+    use_event_durations - optional; if True assigns a duration to each event.
+
+    Returns
+    -------
+        Path to new edf file
+    """
+    if not edf_path:
+        edf_path = Path(data_dir, 'raw.edf')
+
+    params = load_json_parameters(Path(data_dir, 'parameters.json'),
+                                  value_cast=True)
+    raw_data, _, ch_names, _, sfreq = read_data_csv(
+        Path(data_dir, params['raw_data_name']))
+    durations = trigger_durations(params) if use_event_durations else {}
+
+    with open(Path(data_dir, params['trigger_file_name']), 'r') as trg_file:
+        triggers = read_triggers(trg_file)
```

Comment:
We don't correct the offset here, right? I may be following wrong. We should add an argument to handle any offset correction (if present)

Reply:
`read_triggers` is a new helper function that does correct the offset for the returned values. I can add a parameter to be able to return the values uncorrected, but I'm not sure if this would be useful without also providing the offset. Are there any instances in our code when we want these values without the offset correction?

Reply:
We may want to provide it as an argument because there are both system and static offsets. It may be useful to have them provide it as argument. As long as it's handled though we can ask the group during the demo how useful that may or not be.